idempotency_key/locks.py

```diff
@@ -0,0 +1,28 @@
+import abc
+import threading
+
+from idempotency_key.utils import get_lock_timeout
+
+
+class IdempotencyKeyLock(object):
+    @abc.abstractmethod
+    def acquire(self, *args, **kwargs) -> bool:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def release(self):
+        raise NotImplementedError()
+
+
+class SingleProcessLock(IdempotencyKeyLock):
```

Comment:
Instead of naming the class to when it should be used, I think it should be named as what it does like `ThreadLock`.
What do you think?

Reply:
It's also a process lock depending on the class which was why I just said 'lock'. Personally, I don't mind so will let you choose.

Reply:
How about STORAGE_LOCK_CLASS instead? That's what we are using the lock for.

Reply:
Changed to ThreadLock :+1: